### PR TITLE
NumberBox: float values with percent format should be rounded correctly (refix T1093736)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: js/localization/ldml/number.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import { fitIntoRange } from '../../core/utils/math';
+import { fitIntoRange, multiplyInExponentialForm } from '../../core/utils/math';
 import { toFixed } from '../utils';
 
 const DEFAULT_CONFIG = { thousandsSeparator: ',', decimalSeparator: '.' };
 const ESCAPING_CHAR = '\'';
 const MAXIMUM_NUMBER_LENGTH = 15;
+const PERCENT_SHIFT = 2; // '1e2'
```
Comment (ksercs) [inline js/localization/ldml/number.js]:
```suggestion
const PERCENT_EXPONENT_SHIFT = 2; // '1e2'
```

File: js/core/utils/math.js [hunk 2]
```diff
@@ -26,6 +26,24 @@ function getExponent(value) {
...
+    const mantissa = parseFloat(parts[0]);
+    const exponent = parseInt(parts[1]);
+
+    return {
+        exponent,
+        mantissa,
+    };
+}
+
+function multiplyInExponentialForm(value, shift) {
```
Comment (ksercs) [inline js/core/utils/math.js]:
```suggestion
function multiplyInExponentialForm(value, exponentShift) {
```

Comment (EugeniyKiyashko) [reply js/core/utils/math.js]: https://github.com/DevExpress/DevExtreme/pull/22063/commits/8d4e405dd2f10fdd4419e08aa2d6a4aec976da04

File: js/localization/ldml/number.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply js/localization/ldml/number.js]: https://github.com/DevExpress/DevExtreme/pull/22063/commits/8d4e405dd2f10fdd4419e08aa2d6a4aec976da04

**Review by Zedwag — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**